functional/connectivity-loss_test.go

```diff
@@ -0,0 +1,229 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package functional
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/coreos/fleet/functional/platform"
+	"github.com/coreos/fleet/functional/util"
+)
+
+// Check that units states do not change on loss of connectivity to etcd.
+//
+// Note: this only tests the behaviour of the disconnected node;
+// but not the reaction of the rest of the cluster,
+// nor reconciliation after connectivity is restored.
+func TestSingleNodeConnectivityLoss(t *testing.T) {
+	cluster, err := platform.NewNspawnCluster("smoke")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cluster.Destroy()
+
+	m0, err := cluster.CreateMember()
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = cluster.WaitForNMachines(m0, 1)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Set up some units.
+	stateMapping := map[string]struct {
+		command          []string
+		runState         string
+		systemdFileState string
+		systemdState     []string
+	}{
+		"inactive": {[]string{"submit"}, "", "", nil},
+		"loaded":   {[]string{"load", "--no-block"}, "inactive", "enabled", nil},
+		"launched": {[]string{"start", "--no-block"}, "active", "enabled", []string{"loaded", "active", "running"}},
+	}
+	createUnits := map[string][]string{}
+	expectedUnitFiles := map[string]string{}
+	expectedUnitStates := map[string]string{}
+	expectedSystemdFiles := map[string]string{}
+	expectedSystemdStates := map[string][]string{}
+	for _, service := range []string{"single", "global"} {
+		for state, mapping := range stateMapping {
+			unitName := fmt.Sprintf("%s@%s.service", service, state)
+			unitPath := fmt.Sprintf("fixtures/units/%s", unitName)
+			createUnits[unitName] = append(mapping.command, unitPath)
+
+			expectedUnitFiles[unitName] = state
+
+			if mapping.runState != "" {
+				expectedUnitStates[unitName] = mapping.runState
+			}
+
+			if mapping.systemdFileState != "" {
+				expectedSystemdFiles[unitName] = mapping.systemdFileState
+			}
+
+			if mapping.systemdState != nil {
+				expectedSystemdStates[unitName] = mapping.systemdState
+			}
+		}
+	}
+	for name, command := range createUnits {
+		stdout, stderr, err := cluster.Fleetctl(m0, command...)
+		if err != nil {
+			t.Fatalf("Failed creating unit %s: %v\nstdout: %s\nstderr:%s", name, err, stdout, stderr)
+		}
+	}
+
+	checkExpectedStates := func() (isExpected bool, expected, actual map[string]string) {
+		// First check unit files.
+		// These shouldn't change at all after intital submit -- but better safe than sorry...
+		stdout, _, err := cluster.Fleetctl(m0, "list-unit-files", "--no-legend", "--full", "--fields", "unit,dstate")
+		stdout = strings.TrimSpace(stdout)
+		if err != nil {
+			t.Errorf("Failed listing unit files: %v", err)
+		}
+
+		lines := strings.Split(stdout, "\n")
+		actualUnitFiles := map[string]string{}
+		if stdout != "" {
+			for _, line := range lines {
+				cols := strings.Split(line, "\t")
```

Comment:
You can use [Fields](https://golang.org/pkg/strings/#Fields) instead of `Split`.